realm/realm-library/src/main/java/io/realm/internal/PendingRow.java

```diff
@@ -236,9 +239,10 @@ private void notifyFrontEnd() {
                 // No row matches the query, return a invalid row.
                 frontEnd.onQueryFinished(InvalidRow.INSTANCE);
             }
+        } else {
+            clearPendingCollection();
```

Comment:
The naming of these variables / classes could use a refactor -> `Collection` -> `OsResults` and `clearPendingCollection()` -> `clearPendingOsResults()` but that is being tracked elsewhere